cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationOutputStream.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import org.apache.hadoop.fs.Syncable;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.Flushable;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Location outputstream used by {@link LocationManager}
+ */
+public class LocationOutputStream implements Flushable, Closeable {
```

Comment:
On the other hand, you can actually make this class an `OutputStream` by extending from `FilterOutputStream`, which I think is a better approach